web/elm/src/TopBar/TopBar.elm

```diff
@@ -536,24 +491,60 @@ viewMiddleSection model =
                 ]
             ]
 
-        SearchBar r ->
-            viewSearch r model
+        SearchBar ->
+            viewSearch model
 
         Breadcrumbs r ->
             [ Html.div [ id "breadcrumbs", style Styles.breadcrumbContainer ] (viewBreadcrumbs r) ]
 
 
-viewSearch : { query : String, dropdown : Dropdown } -> Model -> List (Html Msg)
-viewSearch r model =
+middleSection : Model r -> MiddleSection
+middleSection { route, query, dropdown, screenSize, groups } =
+    case route of
+        Routes.Dashboard { searchType } ->
+            case
+                ( searchType
+                , query
+                , dropdown
+                , screenSize
+                , groups |> List.concatMap .pipelines
+                )
+            of
+                ( _, _, _, _, [] ) ->
```

Comment:
Ungh this is an ugly case statement. There's got to be a better way to express it... but I can't think of one. Eurgh.